### ISSUE TITLE
[Feature Request] support for DX7II/DX7IIFD patches

### ISSUE THREAD

**Issue by SamuelSHrk** [asb2m10/dexed#165]
Hi, I mentioned this on KVR Audio but I wanted to also request it here. I've noticed most of the time in some of the "All The Web" patches, they become corrupted which would work on FM8 otherwise. I'm curious if the Dexed will be updated to support Mk. 2 sounds and specs since most FM synth emulations only support original DX7 patches.

**Comment by asb2m10** [comment]
Can you pin-point to a specific mk2 cartridge so I can have a look at it ? I'm curious to see how corrupted they are.

**Comment by SamuelSHrk** [comment]
Sure! Let me screenshot some!

**Comment by SamuelSHrk** [comment]
![2018-10-09](https://user-images.githubusercontent.com/43790833/46708618-8a326600-cc0d-11e8-8c08-9e4756b6ce1e.png)

**Comment by SamuelSHrk** [comment]
Also according to this [video](https://www.youtube.com/watch?v=7z-xjCdoWZ4&t=4s), according to the author, this patch only works with DX7 Mk.2 synths because Dexed doesnt have "Poly Unison".

**Comment by SamuelSHrk** [comment]
[
![2018-10-10 1](https://user-images.githubusercontent.com/43790833/46741633-71f53200-cc73-11e8-9309-548ac6a04c18.png)
![2018-10-10](https://user-images.githubusercontent.com/43790833/46741637-74f02280-cc73-11e8-83a3-029ecb7ec047.png)
![2018-10-10 2](https://user-images.githubusercontent.com/43790833/46741642-77eb1300-cc73-11e8-994b-39be06976ea1.png)


](url)

**Comment by SamuelSHrk** [comment]
I also noticed that there is no modulation unlike on the FM8 but that is another issue for another time

**Comment by SamuelSHrk** [comment]
> 
> 
> Can you pin-point to a specific mk2 cartridge so I can have a look at it ? I'm curious to see how corrupted they are.

hi did you get my screenshots?

**Comment by asb2m10** [comment]
Yes, there is extra bytes on some of the TX802 sysex that dexed seems to parse wrongly, I will see what I can do.

**Comment by SamuelSHrk** [comment]
> 
> 
> Yes, there is extra bytes on some of the TX802 sysex that dexed seems to parse wrongly, I will see what I can do.

Thanks.

**Comment by ipogeorge** [comment]
Hi to all. I just started using stand alone Dexed on a win 10 machine. 
My DX7 mk1 is upgraded with the SER (Special Edition Rom).
I don't know if the patch format of SER is the same as the DX7 Mk2/TX802 one, but I think that it's different from the "normal" DX7 mk1, because the control parameters (aftertouch, modwheel, pitchwheel assignments etc.) are saved inside each patch instead of being saved as global ones. 
The SER (Special Edition Rom) upgrade adds a lot of useful features and solves some bugs, so I would love to use Dexed with a full compatibility to its data format.
Is someone aware of the real differences about these data formats?
Thanks to all of you for this nice work.

**Comment by SamuelSHrk** [comment]
Hi, I want to know the progress on this issue. A beta build would be nice, is there also a Centennial Edition VST in the works as well?

**Comment by elovian** [comment]
It seems like there are a number of issues with many of the DX7II / TX802 files:

**alienazi.syx** is a bundle of sysexs. I extracted the two banks as alienazi-a.syx and alienazi-b.syx and attached as a zip file. Since there are two 32 voice banks in this sysex there is no clean way for dexed to determine which bank to load.
[alienazi.zip](https://github.com/asb2m10/dexed/files/5360584/alienazi.zip)

**TX802_Blank_Voices.syx** also contains performance data but only one voice bank. In this case if dexed simply skipped over all non-voice data it would be able to parse the only bank there, I've attached as another zip:
[TX802_Blank_Voices_voice_data_only.zip](https://github.com/asb2m10/dexed/files/5360585/TX802_Blank_Voices_voice_data_only.zip)

**TX802_Blank_Performances.syx** contains only performance data, not voice data, and is not relevant to dexed.

**TX802BankA1** and **TX802BankA2** do not contain voice data and do not contain sysex header and footer data
**TX802BankB1** and **TX802BankB2** do contain voice data but do not contain sysex header and footer data, they are repaired and attached below:
[tx802bankb-fixed.zip](https://github.com/asb2m10/dexed/files/5360597/tx802bankb-fixed.zip)

It seems that the relevant enhancement would be to skip sysex data until you get to a sysex voice data set, then stop parsing.

**Comment by elovian** [comment]
Interesting, according to this:
http://www.muzines.co.uk/articles/everything-you-ever-wanted-to-know-about-system-exclusive/5722
"headerless files" were a thing. Might be an interesting enhancement to attempt to load as headerless data (still checking bounds) since that was a thing apparently. otherwise I'm making a quick utility that can run through and add sysex header and footer data (which is really the proper thing do to, not really much of a data savings anyway/doesn't ensure message integrity)

**Comment by bladeSk** [comment]
After doing a bit of research, DX7II SysEx patch dumps contain this data:

* Metadata (7 B) - not sure what this is for, it seems to contain the voice bank number (0 = 1-32, 1 = 33-64)
* Fractional key scaling data (16165 B)
* Packed supplemental voice data "AMEM" (1128 B) - additional DX7II voice parameters
* Packed voice data "VMEM" (4104 B) - the standard DX7 mk.1 patches that Dexed supports

The formats are described in the DX7II user guide https://www.manualslib.com/manual/1638937/Yamaha-Dx7-Ii-D.html?page=21#manual

Enhancing the engine to play all the additional features would be a lot of work, but it would be great if Dexed could at least read, write and preserve the additional parameters, so that it's possible to mix and match own DX7II carts.

Here's a dump of the pack-in cart: [DX7IIfd ROM1A.zip](https://github.com/asb2m10/dexed/files/10782404/DX7IIfd.ROM1A.zip)

**Comment by solyarisoftware** [comment]
> It seems like there are a number of issues with many of the DX7II / TX802 files:
> 
> **alienazi.syx** is a bundle of sysexs. I extracted the two banks as alienazi-a.syx and alienazi-b.syx and attached as a zip file. Since there are two 32 voice banks in this sysex there is no clean way for dexed to determine which bank to load. [alienazi.zip](https://github.com/asb2m10/dexed/files/5360584/alienazi.zip)
> 

BTW, I'm very probably the author of alienazy.syx (It's incredible that something I did in maybe 1995 is still on the internet) :-) See: https://solyaris.altervista.org/dx7.html

So alienazy.syx is a DX7II "bank", consisting by the sum of 32 DX7II performance patches, where each performance patch is a sum of 2 DX7 patches + some performance parameters/header, if I well remember.
   
See also the related post:
https://github.com/asb2m10/dexed/discussions/405

giorgio
